clang_tools/util.py

```
@@ -86,13 +86,32 @@ def download_file(url: str, file_name: str, no_progress_bar: bool) -> Optional[s
+        given binary.
     """
-    with urllib.request.urlopen(binary_url + ".sha512sum") as response:
-        return response.read(response.length).decode(encoding="utf-8")
+    # Extract the binary filename from the URL (last path segment)
+    bin_filename = binary_url.rsplit("/", 1)[-1]
+
+    # Build the SHA512SUMS URL (same base URL, different file)
+    base_url = binary_url.rsplit("/", 1)[0]
+    sha_url = f"{base_url}/SHA512SUMS"
+
+    with urllib.request.urlopen(sha_url) as response:
```

Ruff (0.15.20)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)